src/DependencyInjection/CsaGuzzleExtension.php

```diff
@@ -113,16 +115,25 @@ private function processCacheConfiguration(array $config, ContainerBuilder $cont
     private function processClientsConfiguration(array $config, ContainerBuilder $container, $debug)
     {
         foreach ($config['clients'] as $name => $options) {
-            $client = new Definition($options['class']);
+            if ($parent = $options['extends']) {
+                $client = new DefinitionDecorator($this->getClientServiceId($parent));
```

Comment:
This may not be clear enough. The question can be shortened this way: do we actually need a `DefinitionDecorator` here?